icu4c/source/test/intltest/unitsroutertest.cpp

```diff
@@ -0,0 +1,33 @@
+// © 2020 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html#License
+
+#include "unicode/utypes.h"
+
+#if !UCONFIG_NO_FORMATTING
+
+#include "intltest.h"
+#include "unicode/unistr.h"
+#include "unitsrouter.h"
+
+
+class UnitsRouterTest : public IntlTest {
+  public:
+    UnitsRouterTest() {}
+
+    void runIndexedTest(int32_t index, UBool exec, const char *&name, char *par = NULL);
+
+    void testBasic();
+};
+
+extern IntlTest *createUnitsRouterTest() { return new UnitsRouterTest(); }
+
+void UnitsRouterTest::runIndexedTest(int32_t index, UBool exec, const char *&name, char * /*par*/) {
+    if (exec) { logln("TestSuite UnitsRouterTest: "); }
+    TESTCASE_AUTO_BEGIN;
+    TESTCASE_AUTO(testBasic);
+    TESTCASE_AUTO_END;
+}
+
+void UnitsRouterTest::testBasic() { IcuTestErrorCode status(*this, "UnitsRouter testBasic"); }
```

Comment:
I suggest adding this to your .clang-format:

AllowShortBlocksOnASingleLine: false

This produces good linebreaks for blocks containing complicated code. In my experience, "return;" is allowed in a one-liner, but anything more complex should be on a separate line, and clang-format can't handle this distinction. So I'm now formatting "return" onto separate lines too.

Reply:
That said, this is testing nothing. Add at least one simple unit test?

Reply:
This comment thread wasn't addressed, or the commit wasn't pushed?

Reply:
Marked as unresolved to remind me - I'll add something.